File: category-template-service/src/main/java/com/salman_ecommerce/category_template_service/dto/template/TemplateDto.java
package com.salman_ecommerce.category_template_service.dto.template;

import java.time.LocalDateTime;
import java.util.List;

import com.salman_ecommerce.category_template_service.entity.Attribute.DataType;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TemplateDto {
	private Long id;
	private String name;
	private String description;
	private TemplateDepartmentDto department;
	private TemplateCategoryDto category;
	private List<TemplateAttributeDto> attributes;
    private LocalDateTime createdAt;

	@Data
	@NoArgsConstructor
	@AllArgsConstructor
	public static class TemplateAttributeDto {
		private Long id;
		private String name;
		private DataType dataType;
	}

	@Data
	@NoArgsConstructor
	@AllArgsConstructor
	public static class TemplateDepartmentDto {
		private Long id;
		private String name;
	}

	@Data
	@NoArgsConstructor
	@AllArgsConstructor
	public static class TemplateCategoryDto {
		private Long id;
		private String name;
	}

}

File: product_catalog_service/src/ProductCatalog.Application/Dtos/Template/TemplateDto.cs
using System;
using System.Collections.Generic;

namespace ProductCatalog.Application.Dtos.Template
{
    public class TemplateDto
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public DepartmentDto? Department { get; set; }
        public CategoryDto? Category { get; set; }
        public List<TemplateAttributeDto>? Attributes { get; set; }
        public DateTime CreatedAt { get; set; }
    }

    public class DepartmentDto
    {
        public int Id { get; set; }
        public string? Name { get; set; }
    }

    public class CategoryDto
    {
        public int Id { get; set; }
        public string? Name { get; set; }
    }

    public class TemplateAttributeDto
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public string? DataType { get; set; }
    }
}
